projects/observability/src/shared/components/topology/d3/layouts/graph-layout.ts

```diff
@@ -1,18 +1,22 @@
 import { RenderableTopology, RenderableTopologyNode, TopologyEdge, TopologyNode } from '../../topology';
 
 export class GraphLayout {
-  private readonly levelToNodesMap: Map<number, RenderableTopologyNode[]> = new Map<number, RenderableTopologyNode[]>([
-    [0, []]
-  ]);
+  private levelToNodesMap: Map<number, RenderableTopologyNode[]> = new Map<number, RenderableTopologyNode[]>([[0, []]]);
   private readonly nodeToLevelMap: Map<RenderableTopologyNode, number> = new Map<RenderableTopologyNode, number>();
 
   public layout(topology: RenderableTopology<TopologyNode, TopologyEdge>): void {
+    this.initialize();
     this.findRootNodes(topology);
     this.fillNodeAndLevelMaps();
     this.assignCoordinatesToNodes();
     this.verticallyCenterAlignNodes();
   }
 
+  private initialize(): void {
+    this.levelToNodesMap = new Map([[0, []]]);
```

Comment:
if we are initializing `levelToNodesMap` here then do we need the initialization at L4?

Reply:
I don't think so.